Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -2051,9 +2065,36 @@ extension SourceKitLSPServer {
     originatorUri: DocumentURI,
     languageService: LanguageService
   ) async throws -> Location {
+    // Check if we're already in the target interface with the same module/group/symbol
+    if let referenceDoc = try? ReferenceDocumentURL(from: originatorUri),
+       case .generatedInterface(let interfaceData) = referenceDoc,
```

Comment:
I think you can slightly shorten this

```suggestion
    if case .generatedInterface(let interfaceData) = try? ReferenceDocumentURL(from: originatorUri),
```